thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MetricCorrelationRankingPipeline.java

```diff
@@ -0,0 +1,355 @@
+  private final String targetInput;
+
+  /**
+   * Constructor for dependency injection
+   *
+   * @param outputName pipeline output name
+   * @param inputNames input pipeline names
+   * @param targetInput input pipeline name for target metrics
+   * @param cache query cache
+   * @param metricDAO metric config DAO
+   * @param datasetDAO datset config DAO
+   */
+  public MetricCorrelationRankingPipeline(String outputName, Set<String> inputNames, String targetInput, QueryCache cache, MetricConfigManager metricDAO, DatasetConfigManager datasetDAO) {
+    super(outputName, inputNames);
+    this.targetInput = targetInput;
+    this.cache = cache;
+    this.metricDAO = metricDAO;
+    this.datasetDAO = datasetDAO;
+  }
+
+  /**
+   * Alternate constructor for PipelineLoader
+   *
+   * @param outputName pipeline output name
+   * @param inputNames input pipeline names
+   * @param properties configuration properties ({@code PROP_TARGET_INPUT})
+   */
+  public MetricCorrelationRankingPipeline(String outputName, Set<String> inputNames, Map<String, String> properties) {
+    super(outputName, inputNames);
+    this.metricDAO = DAORegistry.getInstance().getMetricConfigDAO();
+    this.datasetDAO = DAORegistry.getInstance().getDatasetConfigDAO();
+    this.cache = ThirdEyeCacheRegistry.getInstance().getQueryCache();
+
+    if(!properties.containsKey(PROP_TARGET_INPUT))
+      throw new IllegalArgumentException(String.format("Property '%s' required, but not found.", PROP_TARGET_INPUT));
+    this.targetInput = properties.get(PROP_TARGET_INPUT);
+  }
+
+  @Override
+  public PipelineResult run(PipelineContext context) {
+    Set<MetricEntity> candidateMetrics = filterNonBaselineContext(context);
+
+    TimeRangeEntity currentRange = TimeRangeEntity.getContextCurrent(context);
+    TimeRangeEntity baselineRange = TimeRangeEntity.getContextBaseline(context);
+
+    Set<MetricEntity> targetMetrics = new HashSet<>();
+    for(Entity entity : context.getInputs().get(this.targetInput)) {
+      if(entity instanceof MetricEntity)
+        targetMetrics.add((MetricEntity) entity);
+    }
+
+    LOG.info("Processing {} target metrics and {} candidate metrics", targetMetrics.size(), candidateMetrics.size());
+
+    Set<MetricEntity> allMetrics = new HashSet<>();
+    allMetrics.addAll(candidateMetrics);
+    allMetrics.addAll(targetMetrics);
+
+    // generate requests
+    List<RequestContainer> requestList = new ArrayList<>();
+    requestList.addAll(makeRequests(targetMetrics, currentRange, PRE_CURRENT));
+    requestList.addAll(makeRequests(candidateMetrics, currentRange, PRE_CURRENT));
+    requestList.addAll(makeRequests(targetMetrics, baselineRange, PRE_BASELINE));
+    requestList.addAll(makeRequests(candidateMetrics, baselineRange, PRE_BASELINE));
+
+    LOG.info("Requesting {} time series", requestList.size());
+
+    Map<String, RequestContainer> requests = new HashMap<>();
+    for(RequestContainer somethingVerbose : requestList) {
```

Comment:
No but seriously, it's difficult to follow code with e,t,f,c,df etc.. Please change those